protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -83,9 +88,12 @@ protected final void doProcess(M acceptableMessage, Responder responder, ImapSes
     }
 
     protected final void process(M message, Responder responder, ImapSession session) {
-        final ImapCommand command = message.getCommand();
-        final String tag = message.getTag();
+        ImapCommand command = message.getCommand();
+        String tag = message.getTag();
+
+        TimeMetric timeMetric = metricFactory.timer(command.getName());
         doProcess(message, command, tag, responder, session);
+        timeMetric.elapseTimeInMs();
```

Comment:
When I see the function like that "elapseTimeInMs" the side effect is really not obvious, maybe a better name would be better something such as "logElapsedTimeInMs"